File: tests/expected/exceptions.dart
// @dart=3.4
import 'package:sprintf/sprintf.dart';

show() {
  try {
    throw new Exception("foo");
  } on Exception catch (e) {
    print(sprintf("%s", ["caught"]));
  } finally {
    print(sprintf("%s", ["Finally"]));
  }
  try {
    throw new Exception("foo");
  } catch (e) {
    print(sprintf("%s", ["Got it"]));
  }
  try {
    throw new Exception("foo");
  } on Exception catch (e) {
    assert(e.toString().contains("foo"));
  }
}

main(List<String> argv) {
  show();
}

File: tests/expected/exceptions.go
package main

import (
	"fmt"
	"strings"
)

func Show() {
	{
		// try unsupported

		e := fmt.Errorf("foo")
		fmt.Printf("%v\n", "caught")
		// finally unsupported

		fmt.Printf("%v\n", "Finally")
	}
	{
		// try unsupported

		fmt.Printf("%v\n", "Got it")
	}
	{
		// try unsupported

		e := fmt.Errorf("foo")
		if !(strings.Contains(fmt.Sprintf("%v", e), "foo")) {
			panic("assert")
		}
	}
}

func main() {
	Show()
}
